workers/unity/Packages/io.improbable.worker.sdk/CEventTrace.cs

```diff
@@ -0,0 +1,396 @@
+         * Create a new trace item from the memory owned by this storage. The item will be valid as long
+         * as the storage's memory is valid i.e. until the storage is cleared or destroyed.
+         *
+         * The item is initialized by copying the provided item; pass a NULL item argument to create an
+         * item in an uninitialized state.
+         *
+         * Directly creating a Trace_Item object (on the stack or the heap) by other means than calling this
+         * method is discouraged as it will lead to undefined behaviour when passing that item to certain
+         * trace API methods (e.g. SerializeItemToStream).
+         */
+        [DllImport(WorkerLibrary, CallingConvention = CallingConvention.Cdecl,
+            EntryPoint = "Trace_Item_Create")]
+        public static extern Item* ItemCreate(CIO.Storage storage, Item* item);
+
+        /**
+         * Returns a pointer to a thread-local trace item.
+         *
+         * The item is initially uninitialized, but successive calls to this method on the same thread
+         * always returns the same item, which may have been modified by previous usage.
+         */
+        [DllImport(WorkerLibrary, CallingConvention = CallingConvention.Cdecl,
+            EntryPoint = "Trace_Item_GetThreadLocal")]
+        public static extern Item* ItemGetThreadLocal();
+
+        /**
+         * Get the serialized size of the trace item in bytes.
+         *
+         * Note that each call to Trace_GetSerializedItemSize invalidates the internal state necessary to
+         * serialize the previous item. Therefore, you must call SerializeItemToStream with one item
+         * before calling GetSerializedItemSize with the next item.
+         *
+         * Returns 0 on error. You can call GetLastError to get the associated error message,
+         * but it is safe to pass 0 as the size to a subsequent call to SerializeItemToStream.
+         */
+        [DllImport(WorkerLibrary, CallingConvention = CallingConvention.Cdecl,
+            EntryPoint = "Trace_GetSerializedItemSize")]
+        public static extern Uint32 GetSerializedItemSize(Item* item);
+
+        /**
+         * Serialize the given trace item to the stream.
+         *
+         * The size argument must be the result of a call to GetSerializedItemSize with the same item.
+         * Otherwise, behaviour is undefined. It is not necessary to check that GetSerializedItemSize
+         * returned a non-zero item size. Instead, this can be indirectly checked by passing the size to
+         * this method and checking its return value for an error.
+         *
+         * The caller is responsible for ensuring that the provided stream has sufficient remaining capacity
+         * to hold the serialized item.
+         *
+         * Returns 1 on success, 0 on error. Call Io_Storage_GetLastError to get the associated error
+         * message.
+         */
+        // TODO: Check where Io_Storage_GetLastError exists
```

Comment:
TODO?

Reply:
Oh right, I missed this one. This was before the `CIO.cs` file was written. The `GetLastError` function exists in that file now

Reply:
After I created the `CIO.cs` file, I noticed that it doesn't provide a function signature for `Io_Storage_GetLastError`. It does provide a function called `Io_Stream_GetLastError` so I'm not sure if it's something that was removed or not implemented yet.

Reply:
cc @improbable-jonphillips @tbimprob this looks like a docs error?

Reply:
Yes this was a documentation error (should be ` * Returns 1 on success, 0 on error. Call Trace_GetLastError to get the associated error`) it has been fixed from 14.7.1+ https://github.com/improbable/everything/pull/5459/files